pdir/_internal_utils.py

```diff
@@ -46,3 +45,20 @@ def is_bpython() -> bool:
 
 def is_ptpython() -> bool:
     return _get_repl_type() == ReplType.PTPYTHON
+
+
+def get_first_sentence_of_docstring(obj) -> str:
+    """Parse sentence from docstring by first dot.
+    An empty string will be returned if there're no dots.
+    """
+    line = ''
+    docstring = inspect.getdoc(obj)
+    if docstring:
+        joined = ''.join(docstring.split('\n'))
+        try:
+            first_dot_pos = joined.index('.')
+        except ValueError:
+            line = joined
+        else:
+            line = joined[: first_dot_pos + 1]
+    return line
```

Comment:
I'd write null check at the very start

if not docstring:
  return ''

And you can directly return in try except else instead of assigning to line

Reply:
I agree with null checking in the begining, although I'd like to keep `returns` as few as possible for readability. I'll refine a bit.